services/index.js

```diff
@@ -1,20 +1,31 @@
 'use strict'
 
 const glob = require('glob')
+const { BaseService } = require('./base')
 
 function loadServiceClasses() {
   // New-style services
-  const services = glob
-    .sync(`${__dirname}/**/*.service.js`)
-    .map(path => require(path))
+  const servicePaths = glob.sync(`${__dirname}/**/*.service.js`)
 
   const serviceClasses = []
-  services.forEach(service => {
-    if (typeof service === 'function') {
-      serviceClasses.push(service)
+  servicePaths.forEach(path => {
+    const module = require(path)
+    if (!module) {
+      throw Error(
+        `Expected ${path} to export a service or a collection of services; got ${module}`
+      )
+    } else if (module.prototype instanceof BaseService) {
+      serviceClasses.push(module)
     } else {
-      for (const serviceClass in service) {
-        serviceClasses.push(service[serviceClass])
+      for (const key in module) {
```

Comment:
This assumes that if we've got something that isn't `instanceof BaseService` we've got an iterable, but if I declare a service file that starts

```js
module.exports = class MyService extends SomethingThatDoesntExtendBaseService...
```

we'll get to here and try to iterate `MyService`, which doesn't make much sense.

Reply:
That's true, though is not extending BaseService a use pattern we need to support?

Reply:
No - we don't want to _support_ it. We want to throw an error. That's the point of testing against `(serviceClass.prototype instanceof BaseService)` - right?

but because `MyService` isn't iterable we never enter the loop so nothing ever gets tested against the condition `(serviceClass.prototype instanceof BaseService)` in this case.

Reply:
You'd still want to continue checking as you go, or also throw if

`num valid services != num total services`

to also cover the condition where >=1 services extend `BaseService` and >=1 don't, but yeah.. if we get to the end of that loop and have 0 valid services that's definitely an error.

Reply:
We're right now supporting both arrays and objects, and if we keep doing that, it seems simpler to bail if we ever find a non-service, and check that each file has contained at least one before moving on to the next one.

Alternatively we could drop support for objects and require a single service or an array. Then we could use `count`, validate in `map`, and probably simplify this.

Reply:
In the relatively small subset of services we've refactored so far, we've already found valid uses for multiple export styles:

object:
https://github.com/badges/shields/blob/67193ce09b0c2bc69bb4f0175e6148e942a51e99/services/apm/apm.service.js#L122-L126

array:
https://github.com/badges/shields/blob/67193ce09b0c2bc69bb4f0175e6148e942a51e99/services/npm/npm-downloads.service.js#L94

class:
https://github.com/badges/shields/blob/67193ce09b0c2bc69bb4f0175e6148e942a51e99/services/clojars/clojars.service.js#L13

All of those are sensible module export styles and I think we should continue to support all of them so we keep things relatively flexible for future contributors and allow ourselves and contributors to write idiomatic code. Even if that complicates the service loading code a bit, I think that's worthwhile.

We can detect a module which exports a single class from one exporting an Array or Object using `instanceof Function`:

```
> class FooBar {
... }

> console.log(FooBar instanceof Function);
true

> console.log({} instanceof Function);
false

> console.log([] instanceof Function);
false
```

and special-case it.